lib/client/identityfile/identity.go

```diff
@@ -55,48 +56,65 @@ const (
 	DefaultFormat = FormatFile
 )
 
-// Write takes a username + their credentials and saves them to disk
-// in a specified format.
-//
-// clusterAddr is only used with FormatKubernetes.
-//
-// filePath is used as a base to generate output file names; these names are
-// returned in filesWritten.
-func Write(filePath string, key *client.Key, format Format, clusterAddr string) (filesWritten []string, err error) {
-	const (
-		// the files and the dir will be created with these permissions:
-		fileMode = 0600
-		dirMode  = 0700
-	)
-
-	if filePath == "" {
+const (
+	// The files created by Write will have these permissions.
+	writeFileMode = 0600
+)
+
+// WriteConfig holds the necessary information to write an identity file.
+type WriteConfig struct {
+	// OutputPath is the output path for the identity file. Note that some
+	// formats (like FormatOpenSSH and FormatTLS) write multiple output files
+	// and use OutputPath as a prefix.
+	OutputPath string
+	// Key contains the credentials to write to the identity file.
+	Key *client.Key
+	// Format is the output format for the identity file.
+	Format Format
+	// KubeProxyAddr is the public address of the proxy with its kubernetes
+	// port. KubeProxyAddr is only used when Format is FormatKubernetes.
+	KubeProxyAddr string
+	// OverwriteDestination forces all existing destination files to be
+	// overwritten. When false, user will be prompted for confirmation of
+	// overwite first.
+	OverwriteDestination bool
+}
+
+// Write writes user credentials to disk in a specified format.
+// It returns the names of the files successfully written.
+func Write(cfg WriteConfig) (filesWritten []string, err error) {
+
+	if cfg.OutputPath == "" {
 		return nil, trace.BadParameter("identity output path is not specified")
 	}
 
-	switch format {
+	switch cfg.Format {
 	// dump user identity into a single file:
 	case FormatFile:
-		filesWritten = append(filesWritten, filePath)
-		f, err := os.OpenFile(filePath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, fileMode)
+		if err := checkOverwrite(cfg.OutputPath, cfg.OverwriteDestination); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		filesWritten = append(filesWritten, cfg.OutputPath)
+		f, err := os.OpenFile(cfg.OutputPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, writeFileMode)
```

Comment:
I understand this is existing code being restructured but it would make sense to reuse `writeFile` here as well (which means formatting the output into the buffer first). Might help if we decide to use atomic file writes at some point.